homeassistant/components/here_weather/sensor.py

```diff
@@ -0,0 +1,125 @@
+"""Support for the HERE Destination Weather service."""
+import logging
+from typing import Dict, Optional, Union
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_MODE, CONF_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import Entity
+
+from . import HEREWeatherData
+from .const import DOMAIN, SENSOR_TYPES
+from .utils import convert_unit_of_measurement_if_needed, get_attribute_from_here_data
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, config_entry: ConfigEntry, async_add_entities
+):
+    """Add here_weather entities from a config_entry."""
+    here_weather_data = hass.data[DOMAIN][config_entry.entry_id]
+
+    sensors_to_add = []
+    for sensor_type in SENSOR_TYPES:
+        if sensor_type == config_entry.data[CONF_MODE]:
+            for weather_attribute in SENSOR_TYPES[sensor_type]:
+                sensors_to_add.append(
+                    HEREDestinationWeatherSensor(
+                        config_entry.data[CONF_NAME],
+                        here_weather_data,
+                        sensor_type,
+                        weather_attribute,
+                    )
+                )
+    async_add_entities(sensors_to_add, True)
+
+
+class HEREDestinationWeatherSensor(Entity):
+    """Implementation of an HERE Destination Weather sensor."""
+
+    def __init__(
+        self,
+        name: str,
+        here_data: HEREWeatherData,
+        sensor_type: str,
+        weather_attribute: str,
+        sensor_number: int = 0,  # Additional supported offsets will be added in a separate PR
+    ) -> None:
+        """Initialize the sensor."""
+        self._base_name = name
+        self._name_suffix = SENSOR_TYPES[sensor_type][weather_attribute]["name"]
+        self._here_data = here_data
+        self._sensor_type = sensor_type
+        self._sensor_number = sensor_number
+        self._weather_attribute = weather_attribute
+        self._unit_of_measurement = convert_unit_of_measurement_if_needed(
+            self._here_data.units,
+            SENSOR_TYPES[sensor_type][weather_attribute]["unit_of_measurement"],
+        )
+
+    @property
+    def name(self) -> str:
+        """Return the name of the sensor."""
+        return f"{self._base_name} {self._name_suffix}"
+
+    @property
+    def unique_id(self):
+        """Set unique_id for sensor."""
+        return self.name
```

Comment:
Name is configurable. It's not allowed as unique_id. That must be both unique and stable.